### PR TITLE
Todos of issue #224

### PR DESCRIPTION
**Pull request by bahdotsh**

### CODE REVIEW

File: modules/identities/internal/mappable/identity.go [hunk 1]
```diff
@@ -59,3 +61,55 @@ func NewIdentity(id types.ID, immutableProperties types.Properties, mutablePrope
 		HasMutables:   baseTraits.HasMutables{Properties: mutableProperties},
 	}
 }
+
+func (identity identity) IsProvisioned(address sdkTypes.AccAddress) bool {
```
Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:65]: ## Code Styling
### Interface Implementation
* Code should be ordered as:
```go
type struct foo{}

var _ Foo=(*foo)(nil)

func (foo foo) bar1(){}
func (foo foo) bar1(){}
func (foo foo) bar2(){}

func baz(){}

// Baz
func Baz(){}
```
* The methods implemented should be in the same order as that defined in the interface definition.

Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:65]: add these methods to interIdentity interface also

File: schema/mappables/interIdentity.go [hunk 2]
```diff
@@ -6,12 +6,17 @@
...
 import (
+	sdkTypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/persistenceOne/persistenceSDK/schema/helpers"
 	"github.com/persistenceOne/persistenceSDK/schema/types"
 )
 
 type InterIdentity interface {
 	GetAuthentication() types.Property
 	GetExpiry() types.Property
-
```
Comment (bahdotsh) [inline schema/mappables/interIdentity.go:15]: https://github.com/persistenceOne/persistenceSDK/pull/230/files#r640574415

This is it right?

File: schema/mappables/interIdentity.go [hunk 3]
```diff
@@ -6,12 +6,17 @@
...
+	IsProvisioned(address sdkTypes.AccAddress) bool
```
Comment (bahdotsh) [inline schema/mappables/interIdentity.go:17]: https://github.com/persistenceOne/persistenceSDK/pull/230/files#r640574415

This is it right?

File: modules/identities/internal/mappable/identity.go [hunk 4]
```diff
@@ -59,3 +61,55 @@ func NewIdentity(id types.ID, immutableProperties types.Properties, mutablePrope
...
+	flag := false
```
Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:66]: remove

File: modules/identities/internal/mappable/identity.go [hunk 5]
```diff
@@ -59,3 +61,55 @@ func NewIdentity(id types.ID, immutableProperties types.Properties, mutablePrope
...
+
+	if !ok {
+		panic(errors.IncorrectFormat)
+	}
+
+	if address.Empty() && !accAddressListData.IsPresent(base.NewAccAddressData(address)) {
+		flag = true
+	}
+
+	return flag
```
Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:77]: return false

File: modules/identities/internal/mappable/identity.go [hunk 6]
```diff
@@ -59,3 +61,55 @@ func NewIdentity(id types.ID, immutableProperties types.Properties, mutablePrope
...
+func (identity identity) IsProvisioned(address sdkTypes.AccAddress) bool {
+	flag := false
+	accAddressListData, ok := identity.GetAuthentication().GetFact().(types.ListData)
+
+	if !ok {
+		panic(errors.IncorrectFormat)
+	}
+
+	if address.Empty() && !accAddressListData.IsPresent(base.NewAccAddressData(address)) {
+		flag = true
```
Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:74]: return true

File: modules/identities/internal/mappable/identity.go [hunk 7]
```diff
@@ -59,3 +61,55 @@ func NewIdentity(id types.ID, immutableProperties types.Properties, mutablePrope
 		HasMutables:   baseTraits.HasMutables{Properties: mutableProperties},
 	}
 }
+
+func (identity identity) IsProvisioned(address sdkTypes.AccAddress) bool {
+	flag := false
+	accAddressListData, ok := identity.GetAuthentication().GetFact().(types.ListData)
```
Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:65]: getFact.AsAccAddressList

File: modules/identities/internal/mappable/identity.go [hunk 8]
```diff
@@ -59,3 +61,55 @@ func NewIdentity(id types.ID, immutableProperties types.Properties, mutablePrope
...
+	if !ok {
+		panic(errors.IncorrectFormat)
+	}
+
+	if address.Empty() && !accAddressListData.IsPresent(base.NewAccAddressData(address)) {
+		flag = true
+	}
+
+	return flag
+}
```
Comment (deepanshutr) [inline modules/identities/internal/mappable/identity.go:76]: apply comments in this method every other method in this file

File: schema/mappables/interIdentity.go (hunk 3 above)
Comment (deepanshutr) [reply schema/mappables/interIdentity.go:17]: correct

File: modules/identities/internal/mappable/identity.go (hunk 8 above)
Comment (bahdotsh) [reply modules/identities/internal/mappable/identity.go:76]: Ok

**Review by deepanshutr — APPROVED** [review]
Reviewed.

**Pull request merged**